draft-peabody-dispatch-new-uuid-format-03.txt

```diff
@@ -528,1005 +575,748 @@ Internet-Draft               new-uuid-format               February 2022
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |    var_ver    |            rand_b                             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                            rand_b                             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
+                   Figure 4: UUIDv7 Field and Bit Layout
 
-   unixts:
-      36 bit big-endian unsigned Unix Timestamp value
+   unix_ts_ms:
+      44 bit big-endian unsigned number of milliseconds since midnight 1
+      January 1970 UTC.  Follows the same rules as a UNIX timestamp,
+      i.e. does not include leap seconds.
 
-   subsec_a:
-      12 bits allocated to sub-second precision values.
+   rand_a:
+      20 bits pseudo-random data to provide uniqueness as per
+      Section 5.8
 
-   ver:
-      The 4 bit UUIDv7 version (0111)
+   var_ver:
+      The 8 bit combined variant and version field with the value 0xE7
+      as defined by Section 4.1.
 
-   subsec_b:
-      12 bits allocated to sub-second precision values.
 
-   var:
-      2 bit UUID variant (10)
 
-   subsec_seq_node:
-      The remaining 62 bits which MAY be allocated to any combination of
-      additional sub-second precision, sequence counter, or pseudo-
-      random data.
 
-4.4.1.  UUIDv7 Timestamp Usage
-
-   UUIDv7 utilizes a 36 bit big-endian unsigned Unix Timestamp value
-   (number of seconds since the epoch of 1 Jan 1970, leap seconds
-   excluded so each hour is exactly 3600 seconds long).  The 36 bit
-   value was selected in order to provide more available time to the
-   unix timestamp and avoid the Year 2038 problem by extending the
-   maximum timestamp to the year 4147.
-
-   To achieve a 36 bit UUIDv7 timestamp, the lower 36 bits of a 64 bit
-   unix time are extracted verbatim into UUIDv7
-
-   In the event that 32 bit Unix Timestamp are in use; four zeros MUST
-   be appended at the start in the most significant (left-most) bits of
-   the 32 bit Unix timestamp creating the 36 bit Unix timestamp.  This
-   ensures sorting compatibility with 64 bit unix timestamp which have
-   been truncated to 36 bits.
+Peabody & Davis          Expires 26 August 2022                [Page 11]
+
+Internet-Draft               new-uuid-format               February 2022
 
-   Additional sub-second precision (millisecond, nanosecond,
-   microsecond, etc) MAY be provided for encoding and decoding in the
-   remaining bits in the layout.
 
-   UUIDv8 SHOULD be used in place of UUIDv7 if an application or
-   implementation does not want to truncate a 64 bit Unix Epoch to the
-   lower 36 bits.
+   rand_b:
+      The final 56 bits of pseudo-random data to provide uniqueness as
+      per Section 5.8 and Section 5.5.
 
+4.4.  UUID Version 8
 
+   UUID version 8 provides an RFC-compatible format for experimental or
+   vendor-specific use cases involving time-based UUIDs.  The only
```

Comment:
Is there a specific reason for restricting this to timestamp-specific cases?

I see v8 as the "do whatever you want with it" sandbox.

```suggestion
   vendor-specific use cases.  The only
```